Missing configurationFormType to AsAttributeType

tidall87 opened the pull request:
The AttributeType service has a configuration_form_type entry but the PHP AsAttributeType attribute does not.

| Q               | A
|-----------------|-----
| Branch?         | 2.0, 2.1
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

------------

Trying to create custom attribute type I get an error trying to use AsAttributeType because it missing the configurationFormType property as the service yaml or xml declaration.

**Working**
```yaml
services:
    App\Attribute\AttributeType\EntityAttributeType:
        tags:
            - { name: sylius.attribute.type, attribute_type: entity, label: 'Entity', form_type: 'App\Form\Type\AttributeType\EntityAttributeType', configuration_form_type: 'App\Form\Type\AttributeType\Configuration\EntityAttributeConfigurationType' }
```

**Error**
```php
<?php

declare(strict_types=1);

namespace App\Attribute\AttributeType;

use App\Form\Type\AttributeType\Configuration\EntityAttributeConfigurationType;
use Sylius\Bundle\AttributeBundle\Attribute\AsAttributeType;
use Sylius\Component\Attribute\AttributeType\AttributeTypeInterface;
use Sylius\Component\Attribute\Model\AttributeValueInterface;
use Symfony\Component\Validator\Context\ExecutionContextInterface;

#[AsAttributeType(
    type: 'entity',
    label: 'Entity',
    formType: \App\Form\Type\AttributeType\EntityAttributeType::class,
    configurationFormType: EntityAttributeConfigurationType::class // Unknown named parameter 'configurationFormType'
)]
final class EntityAttributeType implements AttributeTypeInterface
{
    public const string TYPE = 'entity';

    public function getStorageType(): string
    {
        return AttributeValueInterface::STORAGE_JSON;
    }

    public function getType(): string
    {
        return self::TYPE;
    }

    public function getValue(AttributeValueInterface $attributeValue)
    {
        // Implement logic to fetch values
    }

    public function validate(AttributeValueInterface $attributeValue, ExecutionContextInterface $context, array $configuration): void
    {
        // Implement validate() method.
    }
}

```

GSadee committed:
[Admin] Deprecate unused sylius_admin_customer_orders_statistics route

NoResponseMate committed:
[Admin] Deprecate unused sylius_admin_customer_orders_statistics route (#17667)

| Q               | A
|-----------------|-----
| Branch?         | 2.1
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | yes
| Related tickets | fixes #17640
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.13 branch
 - Features and deprecations must be submitted against the 1.14 branch
- Features, removing deprecations and BC breaks must be submitted
against the 2.0 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->


<!-- This is an auto-generated comment: release notes by coderabbit.ai
-->

## Summary by CodeRabbit

- **Documentation**
- Updated the upgrade guide to inform users about changes in
administrative customer order statistics.
  
- **Chores**
- Deprecated an older administrative route for customer order
statistics, with removal planned for a future release.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

GSadee committed:
[CI] Update upmerge branches GH action to add 2.1

mpysiak committed:
[CI] Update upmerge branches GH action to add 2.1 (#17669)

| Q               | A
|-----------------|-----
| Branch?         | 2.1
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations? | no<!-- don't forget to update the UPGRADE-*.md file
-->
| Related tickets | 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.13 branch
 - Features and deprecations must be submitted against the 1.14 branch
- Features, removing deprecations and BC breaks must be submitted
against the 2.0 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->


<!-- This is an auto-generated comment: release notes by coderabbit.ai
-->

## Summary by CodeRabbit

- **Chores**
- Revised the internal merging process to prioritize our latest release
branches and streamline upcoming releases.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

GSadee committed:
Merge branch '2.0' into 2.1

* 2.0: (50 commits)
  [Docs] Fix links to new Sylius 2 documentation in README
  Update docs
  English punctuation - colon
  [CI] Add allow to fail for some randomly failing JS behats
  [Shop] Allow not setting icons for account menu items
  [Shop] Fix main form validation errors rendering
  [Admin][Taxon] Use custom attribute
  [UI] Fix invalid key setting on adding a collection item
  Skip unnecessary code block editing sylius.security.shop_regex
  Skip non-necessary code block
  [Doc] Adapt cookbook documentation about how to disable localized urls from Sylius 1 to Sylius 2
  (in comments) Remove Operation after processing
  The partial operation definition need to be remove, else APIP will try to process it falsely.
  Don't remove the merged operation in case we want to debug the operations
  Merge instead of replacing and allow multiple api resource from various sources
  Remove duplicated info from the original api resource definition
  Change decoration priority to be just after ExtractorResourceMetadataCollectionFactory
  Rename service id to match class name
  Wrong default value fix
  Add AllowedTypes
  ...

GSadee committed:
Merge branch '2.0' into 2.1

* 2.0:
  [Shop] Add tom select styles
  [Shop] Add ux autocomplete

GSadee committed:
Merge branch '2.0' into 2.1

* 2.0:
  Change application's version to v1.13.12-dev
  Generate changelog for v1.13.11
  Change application's version to v1.13.11
  [Documentation] Update 1.13 version in release cycle
  [CS][DX] Refactor

GSadee committed:
Merge branch '2.0' into 2.1

* 2.0:
  Change application's version to v1.14.4-dev
  Generate changelog for v1.14.3
  Change application's version to v1.14.3

GSadee committed:
Merge branch '2.0' into 2.1

* 2.0:
  Change application's version to v2.0.5-dev
  Generate changelog for v2.0.4
  Change application's version to v2.0.4

GSadee committed:
Merge branch '2.0' into 2.1

* 2.0:
  Change application's version to v2.0.6-dev
  Generate changelog for v2.0.5
  Change application's version to v2.0.5
  Fix autocomplete styles
  [Behat] Refactor PricingElement
  [Behat] Enforce strict NodeElement return type
  [Behat][Shop] Change private keyword to protected in page classes methods
  [Behat][Admin] Change private keyword to protected in page classes methods
  [Behat][Shop] Change private keyword to protected in element classes methods
  [Behat][Product] Change private keyword to protected in element classes methods
  [Behat][Admin] Change private keyword to protected in element classes methods
  [Behat] Remove final keyword from element and page classes
  [Maintenance] Fix invalid default address component prop config
  [API] Fix unit test
  [Admin] Limit the tabler version to 1.0
  [API] Fix contract tests for API Platform 4.1

GSadee committed:
Merge branch '2.0' into 2.1

* 2.0:
  update bunnyshell/workflows to v2

GSadee committed:
[UPMERGE] 2.0 -> 2.1 (#17724)

This PR has been generated automatically.
For more details see
[upmerge_pr.yaml](/Sylius/Sylius/blob/1.13/.github/workflows/upmerge_pr.yaml).

**Remember!** The upmerge should always be merged with using `Merge pull
request` button.

In case of conflicts, please resolve them manually with usign the
following commands:
```
git fetch upstream
gh pr checkout <this-pr-number>
git merge upstream/2.1 -m "Resolve conflicts between 2.0 and 2.1"
```

If you use other name for the upstream remote, please replace `upstream`
with the name of your remote pointing to the `Sylius/Sylius` repository.

Once the conflicts are resolved, please run `git merge --continue` and
push the changes to this PR.

<!-- This is an auto-generated comment: release notes by coderabbit.ai
-->

## Summary by CodeRabbit

- **New Features**
- Enhanced product attribute validations to ensure that options like
"Mug material" are accurately assigned.
- Improved product listing checks for more reliable display of products
in orders.
  
- **Refactor**
- Optimized dynamic collection management for smoother overall
operation.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tidall87 committed:
The AttributeType service has a configuration_form_type entry but the PHP AsAttributeType attribute does not.

GSadee reviewed (commented):
Could you adjust the tests after this change?

GSadee commented on src/Sylius/Bundle/AttributeBundle/Attribute/AsAttributeType.php:
```diff
@@ -22,6 +22,7 @@ public function __construct(
         private string $type,
         private string $label,
         private string $formType,
+        private ?string $configurationFormType = null,
         private int $priority = 0,
```
```suggestion
        private int $priority = 0,
        private ?string $configurationFormType = null,
```
to be backward compatible

GSadee reviewed (commented):
And it could be opened against 1.14 🤔

tidall87 committed:
Order missing configuration_form_type entry into the PHP AsAttributeType attribute.

tidall87 committed:
Add tests for missing configuration_form_type entry into the PHP AsAttributeType attribute.

tidall87 commented:
@GSadee I fix and add tests and i also open a new PR from 1.14
https://github.com/Sylius/Sylius/pull/17755

Commit:
Update Sylius Issue templates

Commit:
Add Sylius Plus info to issue templates

GSadee committed:
Update Issue Templates (#17757)

| Q               | A
|-----------------|-----
| Branch?         | 1.13 and above
| License         | MIT

Inspired by
[Symfony](https://github.com/symfony/symfony/tree/7.3/.github/ISSUE_TEMPLATE).


<!-- This is an auto-generated comment: release notes by coderabbit.ai
-->
## Summary by CodeRabbit

- **New Features**
- Introduced updated, structured templates for submitting bug reports,
documentation issues, and feature requests.
- Added a configuration update that disables blank submissions and
provides clear guidance with contact links for support, security, and
translations.
  
- **Chores**
  - Removed legacy issue templates to streamline the reporting process.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

CoderMaggie committed:
Rename config.yaml to config.yml

GSadee committed:
Rename config.yaml to config.yml (#17758)

| Q               | A
|-----------------|-----
| Branch?         | 2.1
| License         | MIT

GitHub only recognizes `config.yml` (with `.yml` extension) for issue
template configurations.

CoderMaggie committed:
Update config.yml with spaces after emojis

GSadee committed:
Update config.yml with spaces after emojis (#17759)

| Q               | A
|-----------------|-----
| Branch?         | 2.1
| License         | MIT

GSadee committed:
Update spaces next to emojis in GH issue template

GSadee committed:
Update spaces next to emojis in GH issue template (#17760)

| Q               | A
|-----------------|-----
| Branch?         | 2.1
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | continuation of #17759 
| License         | MIT

<!--
 - Bug fixes must be submitted against the 1.13 branch
 - Features and deprecations must be submitted against the 1.14 branch
- Features, removing deprecations and BC breaks must be submitted
against the 2.0 branch
 - Make sure that the correct base branch is set

To be sure you are not breaking any Backward Compatibilities, check the
documentation:

https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

tidall87 committed:
Merge branch 'Sylius:2.1' into fix/as-attribute-type-missing-configuration-form-type

GSadee commented:
This one can be closed as there is another PR to 1.14 and after merging it there, we will upmerge the changes to 2.0 and 2.1 👌🏻
